thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -736,19 +861,31 @@ class CIndexGroupAggregateSlaveActivity : public CIndexReadSlaveBase, implements
             gathered = true;
             try
             {
-                ForEachItemIn(p, partDescs)
+                IKeyManager *_currentManager = currentManager;
+                unsigned p = 0;
+                while (true)
                 {
-                    IKeyManager &keyManager = keyManagers.item(p);
-                    setManager(&keyManager);
-                    while (keyManager.lookup(true))
+                    IKeyManager *keyManager = nullptr;
+                    Owned<IIndexLookup> indexInput = getNextInput(keyManager, p, false);
+                    if (!indexInput)
+                        break;
+                    if (keyManager)
+                        prepareManager(keyManager);
+                    while (true)
                     {
+                        const void *key = indexInput->nextKey();
+                        if (!key)
+                            break;
                         ++progress;
-                        noteStats(keyManager.querySeeks(), keyManager.queryScans());
-                        helper->processRow(keyManager.queryKeyBuffer(), this);
+                        noteStats(indexInput->querySeeks(), indexInput->queryScans());
```

Comment:
Is this call in the correct place?

Reply:
it doesn't really need to be as frequent (but not new), but otherwise I think so.